elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -273,14 +272,10 @@ describe('peerPool', () => {
 
 		it('should throw error if no peers selected', async () => {
 			sandbox.stub(peerPool as any, '_peerSelectForRequest').returns([]);
-			let err;
-			try {
-				await peerPool.request(requestPacket);
-			} catch (error) {
-				err = error;
-			}
 
-			expect(err).to.be.instanceOf(RequestFailError);
+			expect(peerPool.request(requestPacket)).to.be.rejectedWith(
```

Comment:
We need `return` here to avoid unhandled errors.

Reply:
Since it's async, i think `return` is not needed?

Reply:
@shuse2 I thought after our discussion, it was clear that we need it. Please, check it out again. If we change the expectation to another kind of error for example, we obtain an `UnhandledPromiseRejectionWarning: AssertionError` with a false positive.

@mitsujutsu check it out for yourself too. I suggest to include as well the message error like this:
```
			expect(peerPool.request(requestPacket)).to.eventually.be.rejectedWith(
				RequestFailError,
				'Request failed due to no peers found in peer selection',
			);
```